pkg/auth/oauth2authenticator.go

```diff
@@ -359,47 +407,27 @@ func (o *oAuth2Authenticator) authenticateWithAuthorizationCode() error {
 		// launch browser
 		go o.openBrowserFunc(authCodeUrl)
 
-		timedOut := false
-		timer := time.AfterFunc(TIMEOUT_SECONDS, func() {
-			timedOut = true
+		// Handle context cancellation & timeout
+		go func() {
+			<-ctx.Done()
+			// We impose that o.shutdownServerFunc is/calls a server function that is safe to be called before srv.Server
+			// and will result in srv.Server exiting immediately when called.
 			o.shutdownServerFunc(srv)
-		})
+		}()
+
 		listenErr = srv.Serve(listener)
 		if errors.Is(listenErr, http.ErrServerClosed) { // if the server was shutdown normally, there is no need to iterate further
-			if timedOut {
-				return errors.New("authentication failed (timeout)")
+			switch err := ctx.Err(); {
+			case errors.Is(err, context.Canceled):
+				return false, nil // No need to error, the user canceled this auth request
```

Comment:
Question: Shouldn't we return an error here instead of swallowing it in the library? If the application decides to swallow it, it can but from a framework perspective we shouldn't take the decision.